x/concentrated-liquidity/types/errors.go

```diff
@@ -913,3 +913,11 @@ type LastPositionTransferError struct {
 func (e LastPositionTransferError) Error() string {
 	return fmt.Sprintf("cannot transfer position %d because it is the last position in pool %d", e.PositionId, e.PoolId)
 }
+
+type ContractHookOutOfGasError struct {
+	GasLimit uint64
+}
+
+func (e ContractHookOutOfGasError) Error() string {
+	return fmt.Sprintf("A single contract call cannot exceed %d gas in a CL hook call.", e.GasLimit)
```

Comment:
Should we also format how much gas it actually took?

Reply:
Hmm I don't think this is possible since we would exit once the limit is hit (so the gas consumed would be the limit amount).

Since the attack vector here is unbounded gas consumption with an infinite loop, it wouldn't make sense to continue metering until execution completes so we know how much to put in a field like this.